src/protocol/get_pfm_id.rs

```diff
@@ -0,0 +1,214 @@
+        w.write_le(self.port_id)?;
+        self.pfm_region.to_wire(&mut w)?;
+        // TODO(asnaider): If `IdentifierType::Version` do we skip to save bytes?
+        self.identifier.to_wire(&mut w)?;
+        Ok(())
+    }
+
+    struct Response<'wire> {
+        /// PFM validity.
+        pub pfm_valid: PFMValid,
+        /// PFM Identifier (either Version or Platform).
+        #[cfg_attr(feature = "serde", serde(borrow))]
+        pub pfm_id: Identifier<'wire>,
+    }
+
+    fn Response::from_wire(r, arena) {
+        let pfm_valid = PFMValid::from_wire(r, arena)?;
+        let identifier = IdentifierType::from_wire(r, arena)?;
+        let pfm_id = match identifier {
+            IdentifierType::Version => {
+                let mut version: [u8; 4] = [0; 4];
+                r.read_bytes(&mut version)?;
+                Identifier::Version {id: version}
+            }
+            IdentifierType::Platform => {
+                if r.remaining_data() == 0 {
+                    Identifier::Platform {id: &[]}
+                } else {
+                    let bytes = r.read_direct(
+                    arena,
+                    Layout::from_size_align(
+                        r.remaining_data(),
+                        core::mem::align_of::<u8>())
+                    .expect(
+                        "Layout for u8 slice can't be created. This indicates a bug in the code."))?;
+
+                    Identifier::Platform{id: bytes}
+                }
+            }
+        };
+        Ok(Self { pfm_valid, pfm_id })
+    }
+
+    fn Response::to_wire(&self, w) {
+        self.pfm_valid.to_wire(&mut w)?;
+        match self.pfm_id {
+            Identifier::Version{id: ref bytes } => {
+                let idtype = IdentifierType::Version;
+                idtype.to_wire(&mut w)?;
+                w.write_bytes(bytes)?;
+            }
+            Identifier::Platform{ id: ref bytes } => {
+                let idtype = IdentifierType::Platform;
+                idtype.to_wire(&mut w)?;
+                w.write_bytes(bytes)?;
+            }
+        }
+        Ok(())
+    }
+
+}
+
+wire_enum! {
+    /// PFM Region.
+    #[cfg_attr(feature = "arbitrary-derive", derive(Arbitrary))]
+    #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+    pub enum PFMRegion: u8 {
+        /// PFM Region Active
+        Active = 0x00,
+        /// PFM Region Pending
+        Pending = 0x01,
+    }
+}
+
+wire_enum! {
+    /// The id type requested.
+    #[cfg_attr(feature = "arbitrary-derive", derive(Arbitrary))]
+    #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+    pub enum IdentifierType: u8 {
+        /// PFM Version ID
+        Version = 0x00,
+        /// PFM Platform ID
+        Platform = 0x01,
+    }
+}
+
+wire_enum! {
+    /// The id type requested.
+    #[cfg_attr(feature = "arbitrary-derive", derive(Arbitrary))]
+    #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+    pub enum PFMValid: u8 {
```

Comment:
TBH I feel like a `bool` would have been a more appropriate type here. WDYT?

Reply:
That's fair, but having played around with it, it's more ergonomic to me to have the enum as is since I can encode the values of valid/invalid directly as opposed to having to set up the from/to wire.